### PR TITLE
Implement Base58 Encode and Decode Interops on Neo

### PR DESCRIPTION
**Pull request by meevee98**
Included two new interops:
- `System.Binary.Base58Encode`
- `System.Binary.Base58Decode`

### CODE REVIEW

**Comment by erikzhang** [comment]
I will close this because it has been discussed and be denied in #263.

**Comment by meevee98** [comment]
@erikzhang #263 is about the address conversion to script hash, that takes some other validations besides the base58 conversion.
In #1614 base58 syscall is listed as a desired feature

**Comment by Tommo-L** [comment]
`Base58` is not only used for addresses, but also can be used for contract. For example, one contract request a url, which returns base58 value.

File: src/neo/SmartContract/ApplicationEngine.Binary.cs [hunk 1]
```diff
@@ -29,5 +32,15 @@ protected internal byte[] Base64Decode(string s)
         {
             return FromBase64String(s);
         }
+
+        internal string Base58Encode(byte[] data)
```
Comment (erikzhang) [inline src/neo/SmartContract/ApplicationEngine.Binary.cs:36]: protected

File: tests/neo.UnitTests/SmartContract/UT_ApplicationEngine.cs [hunk 2]
```diff
@@ -26,10 +26,16 @@ public void TestBinary()
             var data = new byte[0];
             CollectionAssert.AreEqual(data, engine.Base64Decode(engine.Base64Encode(data)));
 
+            CollectionAssert.AreEqual(data, engine.Base58Decode(engine.Base58Encode(data)));
```
Comment (shargon) [inline tests/neo.UnitTests/SmartContract/UT_ApplicationEngine.cs:29]: Duplicate with L27?

Comment (meevee98) [reply tests/neo.UnitTests/SmartContract/UT_ApplicationEngine.cs:29]: The logic is the same but the first tests Base64 and the other tests Base58.

**Comment by shargon** [comment]
@meevee98 could you update the base branch?

**Pull request merged**